components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -1420,4 +1427,29 @@ private void setLoggedInOrgIdInRequest(AuthenticatedIdPData authenticatedIdPData
             }
         }
     }
+
+    private void setUserOrganization(AuthenticatedUser authenticatedUser, AuthenticatorConfig authenticatorConfig) {
+
+        String userOrganization = null;
+        if (authenticatedUser.getUserAttributes() != null && isLoggedInWithOrganizationLogin(authenticatorConfig)) {
+            for (Map.Entry<ClaimMapping, String> userAttributes : authenticatedUser.getUserAttributes().entrySet()) {
+                if (FrameworkConstants.USER_ORGANIZATION_CLAIM.equals(
+                        userAttributes.getKey().getLocalClaim().getClaimUri())) {
+                    userOrganization = userAttributes.getValue();
+                }
+            }
+        }
+
+        if (userOrganization == null) {
```

Comment:
```suggestion
        if (StringUtils.isEmpty(userOrganization)) {
```
WDYT?

Reply:
Agree on the suggession. But the implementation was changed bit. Please review the method